### PR TITLE
Sparse jacobian/KLU support

### PR DESCRIPTION
**Pull request by spcornelius**
These changes add support for the linear solver KLU and with it the option to supply a fully sparse (compressed sparse column, CSC) Jacobian.

I have tried to incorporate this into AnyODE in the most generic way possible, and to mirror the current implementation for banded Jacobians/linear solvers within pycvodes. My detailed
thinking can be found in individual commit messages. Nonetheless, I expect there are smarter
ways of doing this...

### CODE REVIEW

File: pycvodes/_cvodes.pyx [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import warnings
 from cpython.object cimport PyObject
-from libc.stdlib cimport malloc
+from libc.stdlib cimport malloc, free
```
Comment (bjodah) [inline pycvodes/_cvodes.pyx]: `free` is an unused import I believe?

**Comment by bjodah** [comment]
This looks good to me after glancing over it. I'll take a closer look tomorrow.

If you want to run the drone tests locally there is a cli program for Linux and macOS here:
http://readme.drone.io/0.4/devs/cli/

and here's the convenience script I use together with it (under linux): 
https://gist.github.com/bjodah/f3706f476a0b8791c99a65f801c906be

**Comment by bjodah** [comment]
Thank you for this, I went ahead and tweaked setup & code until we got green light on Linux, macOS and Windows.

Do you want to add anything else here (maybe the CHANGES.rst?), or should I go ahead and merge this? (I can update the changelog when making the new release).

**Comment by bjodah** [comment]
I went ahead and merged (I realized I want to make a release to test some other dependent projects with these new changes)

**Comment by spcornelius** [comment]
Hi Björn. Sorry, I have been traveling and haven't been able to get back to this. I realize there were some oversights in the version of the branch that you merged. I have made some additional changes to make it pass your original tests (they now all pass, as do some new ones of my own creation, at least when drone'd locally using a copy of your development image)

However I see that you have since changed AnyODE to be an external submodule (probably a good idea!). Let me push some prerequisite changes to there first...

**Comment by bjodah** [comment]
No worries! Hope the trip went well. Yes, I changed it into a proper submodule since it's used in so many different repositories that it was becoming hard to remember to copy the files manually to all the right places. One minor down side, however, is that the output from git-archive is no longer an pip-installable file (since submodules are not included by default), it's really a minor issue though.

**Pull request merged**